usr/share/rear/restore/BAREOS/default/400_restore_backup.sh

```diff
@@ -1,178 +1,114 @@
-you to restore the required files. It's assumed that you know what is necessary
-to restore - typically it will be a full backup.
-
-Do not exit 'bconsole' until all files are restored
-
-WARNING: The new root is mounted under '$TARGET_FS_ROOT'.
-
-Press ENTER to start bconsole"
-        read
-
-        bconsole
+    # bareos_recovery_mode == automatic: restore most recent backup automatically
+    
+    LogPrint "starting restore using bconsole:"
+    LogPrint "$RESTORE_CMD"
+
+    # example output of 'bcommand_json "$RESTORE_CMD"':
+    # {
+    #   "jsonrpc": "2.0",
+    #   "id": null,
+    #   "result": {
+    #     "query": [
+    #       {
+    #         "jobid": "18",
+    #         "level": "F",
+    #         "jobfiles": "59653",
+    #         "jobbytes": "2504970763",
+    #         "starttime": "2024-06-05 13:58:31",
+    #         "volumename": "Full-0001"
+    #       },
+    #        {
+    #         "jobid": "90",
+    #         "level": "I",
+    #         "jobfiles": "6553",
+    #         "jobbytes": "284336784",
+    #         "starttime": "2024-06-18 14:08:40",
+    #         "volumename": "Incremental-0009"
+    #       }
+    #     ],
+    #     "run": {
+    #       "jobid": "103"
+    #     }
+    #   }
+    # }
+    local restore_cmd_result
+    restore_cmd_result="$( bcommand_json "$RESTORE_CMD" )"
+    Log "$restore_cmd_result"
+    if [ -z "$restore_cmd_result" ] || ! restore_jobid="$( jq  --exit-status --raw-output '.result.run.jobid' <<< "$restore_cmd_result" )"; then
+        Error "Failed to determine Restore Job."
     fi
-    LogPrint "
-Please verify that the backup has been restored correctly to '$TARGET_FS_ROOT'
-in the provided shell. When finished, type exit in the shell to continue
-recovery.
-"
+    LogPrint "JobId of restore job is $restore_jobid"
+fi
 
-if [[ "$ISO_RECOVER_MODE" = "unattended" ]] || [[ "$PXE_RECOVER_MODE" = "unattended" ]] ; then
-    Log "Unattended mode selected"
+wait_restore_job "$restore_jobid"
+local job_exitcode=$?
+if [ ${job_exitcode} -eq 0 ]; then
+    Log "$( bcommand "list joblog jobid=$restore_jobid" )"
+    LogPrint "Restore job finished successfully."
+elif [ ${job_exitcode} -eq 1 ]; then
+    Log "$( bcommand "list joblog jobid=$restore_jobid" )"
+    LogPrint "WARNING: Restore job finished with warnings."
 else
-    rear_shell "Did the backup successfully restore to '$TARGET_FS_ROOT' ? Ready to continue ?" \
-            "bls -j -V$BEXTRACT_VOLUME $BEXTRACT_DEVICE
-vi bootstrap.txt
-bextract$exclude_list -b bootstrap.txt -V$BEXTRACT_VOLUME $BEXTRACT_DEVICE $TARGET_FS_ROOT"
+    LogPrint "$( bcommand "list joblog jobid=$restore_jobid" )"
+    Error "Bareos restore failed (${job_exitcode})"
 fi
 
-fi # end of [[ "$BEXTRACT_DEVICE" || "$BEXTRACT_VOLUME" ]]
-
-
-mkdir $TARGET_FS_ROOT/var/lib/bareos && chroot $TARGET_FS_ROOT chown bareos: /var/lib/bareos
+mkdir "$TARGET_FS_ROOT/var/lib/bareos" && chroot "$TARGET_FS_ROOT" chown bareos: /var/lib/bareos
```

Comment:
Sorry for asking again:

`chown bareos $TARGET_FS_ROOT/var/lib/bareos` won't just work? I don't see the need for the `chroot` here.

Reply:
Yes, it does. Thank you.

Reply:
Yes, I already changed it in this way.